packages/manager/apps/web/client/app/hosting/multisite/git-association/hosting-multisite-git-association.service.js

```diff
@@ -0,0 +1,24 @@
+export default class HostingMultisiteGitAssociationService {
+  /* @ngInject */
+  constructor($http) {
+    this.$http = $http;
+  }
+
+  getVcsWebhookUrls(serviceName, path, vcs) {
+    return this.$http
+      .get(`/hosting/web/${serviceName}/vcs/webhooks?path=${path}&vcs=${vcs}`)
+      .then(({ data }) => data);
+  }
+
+  getSshKey(serviceName) {
+    return this.$http
+      .get(`/hosting/web/${serviceName}/key/ssh`)
+      .catch((error) => {
+        if (error.status === 404) {
+          return this.$http.post(`/hosting/web/${serviceName}/key/ssh`);
+        }
+        return error;
+      })
+      .then(({ data }) => data.publicKey);
```

Comment:
`catch` and `then` block should be reversed

Reply:
nope, because the logic is like this (cf ssh key api: first we get, then if error we create, then we retrieve the value of the publicKey)

Reply:
Ok but how about code if code error is not 404 ? catch block will call then block anyway

Reply:
This will be handled in other ticket dedicated to this